GitUI/BranchTreePanel/RepoObjectsTree.cs

```diff
@@ -173,6 +176,11 @@ void CollectFromNodes(TreeNodeCollection nodes)
             }
         }
 
+        public void AddDependencies(ICanGetSelectedRevisions canGetSelectedRevisions)
+        {
+            _canGetSelectedRevisions = canGetSelectedRevisions;
+        }
```

Comment:
This introduces a new pattern, which isn't a problem in itself, however it increases an entropy.
All necessary dependencies to `RepoObjectsTree` are supplied via `Initialize`
https://github.com/gitextensions/gitextensions/blob/521e9e8793799a9b6969868fdfc60f2f5f12a32d/GitUI/BranchTreePanel/RepoObjectsTree.cs#L210-L218

Reply:
Sorry. 😄